### PR TITLE
rewrite settings yml to include configuration description

### PR DESCRIPTION
**Pull request by dmarkiewicz**

### CODE REVIEW

File: acid/features/history/controller.py [hunk 1]
```diff
@@ -14,10 +14,10 @@
 @builds.route('/builds/<int:page>')
 @db_session
 def show_builds_history(page=1):
-    per_page = current_app.config['buildset']['per_page']
+    per_page = current_app.config['history']['pagination']['per_page']
```
Comment (xbend) [inline acid/features/history/controller.py]: Maybe put:
config = current_app.config
And later refer just to config, like in other change in controller.py

File: acid/features/history/controller.py [hunk 2]
```diff
@@ -14,10 +14,10 @@
...
     pipeline = current_app.config['default']['pipename']
-    page_links = current_app.config['buildset']['page_links']
-    buildset_log_url = current_app.config['buildset']['log_url']
+    page_links = current_app.config['history']['pagination']['page_links']
+    buildset_log_url = current_app.config['history']['log_server']
```
Comment (xbend) [inline acid/features/history/controller.py]: Consider changing the variable name to log_server_url. More consistent with config dictionary naming.

File: settings.yml [hunk 3]
```diff
@@ -1,41 +1,115 @@
 ---
 default:
+  # Default pipename for status controller, used when pipename is not provided.
```
Comment (xbend) [inline settings.yml]: Default pipename -> Default pipeline name
Part after comma is unnecessary

File: settings.yml [hunk 4]
```diff
@@ -1,41 +1,115 @@
...
   pipename: 'periodic-nightly'
+
+  # YAML file where admins emails are stored.
```
Comment (xbend) [inline settings.yml]: admins emails -> admin emails

File: settings.yml [hunk 5]
```diff
@@ -1,41 +1,115 @@
...
-  password: 'root'
-  port: 3306
-  name: 'zuul'
-buildset:
-  per_page: 20
-  page_links: 4
-  log_url: 'http://logs.opencontrail.org'
+
+
+# Configuration of authentication.
```
Comment (xbend) [inline settings.yml]: Not a sentence, remove ending dot.

File: settings.yml [hunk 6]
```diff
@@ -1,41 +1,115 @@
...
+  # Specify pipelines which will be visible in UI.
+  # If a pipeline listed here does not exist in zuul, it will not show up.
   pipelines:
     - check
     - gate
     - periodic-nightly
     - periodic-sanity
     - experimental
     - experimental-provision
+    - some-weird-pipeline
```
Comment (xbend) [inline settings.yml]: Remove entries used for personal testing.

File: settings.yml [hunk 7]
```diff
@@ -1,41 +1,115 @@
...
           - master
           - R5.0
+
+  # Data used by PonyORM to bind database object with specyfic database
+  # Read more: https://docs.ponyorm.com/database.html#binding-the-database-object-to-a-specific-database 
+  database:
+    # Name of database provider, which is a module of pony.orm.dbproviders package.
+    provider: 'mysql'
+
+    # IP address of database server.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 8]
```diff
@@ -1,41 +1,115 @@
...
+    host: '10.10.10.5'
+
+    # Database username.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 9]
```diff
@@ -1,41 +1,115 @@
...
+    user: 'root'
+
+    # Database password.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 10]
```diff
@@ -1,41 +1,115 @@
...
+    password: 'root'
+
+    # TCP port of database server.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 11]
```diff
@@ -1,41 +1,115 @@
...
+    port: 3306
+
+    # Database name.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 12]
```diff
@@ -1,41 +1,115 @@
...
+    name: 'zuul'
+  
+  # Configuration of ZuulManager.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings.yml [hunk 13]
```diff
@@ -1,41 +1,115 @@
...
   manager:
+    # IP address of zuul server.
```
Comment (xbend) [inline settings.yml]: Remove ending dot.

File: settings_test.yml [hunk 14]
```diff
@@ -25,17 +25,25 @@ zuul:
     - periodic-sanity
     - experimental
     - experimental-provision
+    - some-weird-pipeline
```
Comment (xbend) [inline settings_test.yml]: Remove entries used for personal testing.

File: settings_test.yml [hunk 15]
```diff
@@ -3,20 +3,20 @@ default:
...
+    launchpad:
+      url: 'https://login.launchpad.net/+openid'
+
+history:
+  log_server: 'http://logs.opencontrail.org/'
+  pagination:
+    per_page: 20
+    page_links: 4
+
 zuul:
```
Comment (xbend) [inline settings_test.yml]: Missing comment blocks for zuul section.

File: settings.yml [hunk 16]
```diff
@@ -1,41 +1,115 @@
...
+    per_page: 20
+
+    # Number of links displayed before or after current page link.
+    page_links: 4
+
+
 zuul:
-  url: 'http://10.10.10.5/'
+  # <protocol>://<host> for zuul server.
+  zuul_server: 'http://10.10.10.5/'
```
Comment (xbend) [inline settings.yml]: Note sure about naming here. Looks weird in code when referred to:
For ex: zuul_url = current_app.config["zuul"]["zuul_server"]
Maybe leave it as 'url'. The format of the value is an url, so 'url' is still quite fitting.

File: settings_test.yml (hunk 15 above)
Comment (xbend) [reply settings_test.yml]: Sorry forgot about this, missed that its for testing settings.

File: acid/features/auth/controller.py [hunk 17]
```diff
@@ -14,8 +14,9 @@
 
 @auth.route('/sign_in')
 def sign_in():
+    config = current_app.config
```
Comment (xbend) [inline acid/features/auth/controller.py]: extract section specific config here, like in dp.py
so:
config = current_app.config['authentication']

File: acid/features/status/controller.py [hunk 18]
```diff
@@ -10,11 +10,12 @@
 @status.route('/status')
 @status.route('/status/<string:pipename>')
 def show_status(pipename=None):
-    zuul_url = current_app.config["zuul"]["url"]
-    zuul_endpoint = current_app.config['zuul']['status_endpoint']
+    config = current_app.config
+    zuul_url = config["zuul"]["url"]
```
Comment (Mlastawi) [inline acid/features/status/controller.py]: Single quotes are used to address a field in a dictionary almost everywhere in code.
Change double quotes to single for consistency.

File: acid/features/status/controller.py [hunk 19]
```diff
@@ -27,11 +28,12 @@ def show_status(pipename=None):
 
 @status.route('/')
 def show_dashboard():
-    zuul_url = current_app.config["zuul"]["url"]
-    zuul_endpoint = current_app.config['zuul']['status_endpoint']
+    config = current_app.config
+    zuul_url = config["zuul"]["url"]
```
Comment (Mlastawi) [inline acid/features/status/controller.py]: Double vs single quotes issue

**Pull request merged**